test/8_custom_logger.sh

```diff
@@ -0,0 +1,29 @@
+. "$(dirname "$0")/functions.sh"
```

Comment:
you probably know more bash/zsh/sh than me...
but the quotes around `"$0"` don't seem right
```sh
. "$(dirname "$0")/functions.sh"
```
wouldn't this produce a syntax error?
```sh
"$(dirname "
```

Reply:
It should be valid because the inner `"$0"` is encased in a `$(...)` command substitution construct.

The [bash documentation](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Command-Substitution) tells that:
> When using the $(command) form, all characters between the parentheses make up the command; none are treated specially.

The only issue occurs when using backquotes in backquotes:
> Command substitutions may be nested. To nest when using the backquoted form, escape the inner backquotes with backslashes.

The [zsh documentation](https://zsh.sourceforge.io/Doc/Release/Expansion.html#Command-Substitution) basically tells the same thing.

For sh POSIX shells, the [documentation](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_03) basically boils down to this: the substituted command is executed in a subshell as is.

I guess that should cover the potential issue :relieved:.